### PR TITLE
feat: Optimize the timing of task run()

### PR DESCRIPTION
**Pull request by lazyky**
-   Change `model_update_time` tag in Stable `training_data` to `train_start_time`
-   Add `update_train_start_time()` to update by timestamp of latest available model
-   Delete `model_update_time` in `Metadata`

### CODE REVIEW

File: cml-tdengine/src/core/register.rs [hunk 1]
```diff
@@ -266,6 +257,15 @@ mod tests {
...
         assert_eq!(4, records);
+        // check tag count
+        let tag_records = taos
+            .query_one(format!(
+                "SELECT COUNT(*) FROM INFORMATION_SCHEMA.INS_TAGS WHERE db_name = 'training_data' and table_name = '{}'",
+                another_metadata_check.batch()
+            ))
+            .await?
+            .unwrap_or(0);
+        assert_eq!(3, tag_records);
```
Comment (bioinformatist) [inline cml-tdengine/src/core/register.rs:268]: Can you explain the goal of this check?

File: cml-tdengine/src/core/task.rs [hunk 2]
```diff
@@ -24,6 +24,30 @@ struct TaskInfo {
     ts: DateTime<Local>,
 }
 
+fn update_train_start_time(taos: &Taos, available_status: &[&str]) {
```
Comment (bioinformatist) [inline cml-tdengine/src/core/task.rs:27]: As the function is only called within the `run()` function, there is no need to abstract it out.

**Review by bioinformatist — CHANGES_REQUESTED** [review]
Totally good job.

File: cml-core/src/core/task.rs [hunk 3]
```diff
@@ -22,6 +22,7 @@ pub trait Task<M> {
     fn run<FN>(
         &self,
         task_config: TaskConfig,
+        available_status: &[&str],
```
[the remark this answers is not in the document]
Comment (lazyky) [reply cml-core/src/core/task.rs:25]: I have thought about this way, but I want to tell the user that the `available_status` in `inference` is the same as the one in `task`.

File: cml-tdengine/src/core/register.rs (hunk 1 above)
Comment (lazyky) [reply cml-tdengine/src/core/register.rs:268]: Check whether the count of labels is inserted correctly by using the code below.
[register.rs#L52-L56](https://github.com/bioinformatist/cml/blob/0b7886f24934d3b6c574d21aadb10c024d98cbf3/cml-tdengine/src/core/register.rs#L52-L56)

File: cml-core/src/core/task.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (lazyky) [reply cml-core/src/core/task.rs:25]: You're right.

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -27,6 +27,7 @@ jobs:
         image: tdengine/tdengine:${{ needs.get-taos-version.outputs.taos-version }}
         ports:
           - 6030:6030
+          - 6041:6041
```
Comment (bioinformatist) [inline .github/workflows/test.yml:30]: There's no need to use `6041` with native connection.

File: cml-tdengine/Cargo.toml [hunk 5]
```diff
@@ -8,7 +8,7 @@ edition = "2021"
 [dependencies]
 cml-core = { path = "../cml-core" }
 
-taos = { version = "*", default-features = false, features = ["native"] }
+taos = { version = "*", default-features = false, features = ["ws"] }
```
Comment (bioinformatist) [inline cml-tdengine/Cargo.toml:11]: Use `native` instead.

**Review by bioinformatist — CHANGES_REQUESTED** [review]
We have already switched back to native feature of `taos`

**Pull request merged**